lib/rubocop/cop/security/json_load.rb

```diff
@@ -0,0 +1,84 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Security
+      # This cop checks for the use of unsecure JSON methods.
+      #
+      # @example
+      #   # always offense
+      #   JSON.load("{}")
+      #   JSON.restore("{}")
+      #
+      #   # no offense
+      #   JSON.parse("{}")
+      #
+      class JSONLoad < Cop
+        DANGEROUS_METHODS = [:load, :restore].freeze
+
+        MSG = 'Do not use `JSON#%s`. Use `JSON#parse instead`.'.freeze
+
+        def on_const(node)
+          mod, klass = *node
+          return unless (mod.nil? || mod.cbase_type?) && method_send?(node)
+
+          check_json_node(klass, node.parent) if klass == :JSON
+        end
+
+        private
+
+        def check_json_node(_, node)
```

Comment:
I'd suggest using a pattern matcher instead. It would simplify the code a lot. Check for usages of `def_node_matcher`.